docs/DeveloperGuide.md

```diff
@@ -7,31 +7,67 @@
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Exception
+### Itemlist
+Itemlist class is an object which contains items to be added to the stock inventory list. Able to add items, remove functions, edit items inside
+the list.
+### Item
+Item class is an object which represents an item in the stock inventory list. Stores data about the item such as item price, 
+quantity of item, and others.
+### Parser
+Parser class processes user inputs and sieves out relevant details before calling the relevant methods.
+It contains command formats that must be adhered to for the methods to be called.
+### Storage
+Storage class contains methods to write description of items to the file `./StockMasterData.txt`, 
+and retrieve information from the file when program restarts.
+### UI
 
 
 ## Product scope
+StockMaster allows users to use the following commands:
+* add: add an item, quantity, category, buy price and sell price
+* del: delete an item
+* edit: edit the quantity of an item
+* sell: sell a certain quantity of an item at a stated price
+* find: find a specific item using a keyword
+* list: list all items in the inventory
+* help: list all commands
+* exit: exit StockMaster
+
```

Comment:
Hopefully, you can explain what the contents of each methods are and how the methods communicate with each related methods. This DG is to teach an interested developer how to contribute to the codebase.